In [1]:
import pandas as pd 
import numpy as np
import matplotlib as plt
import scipy
from scipy import stats
import statsmodels.api as sm
### Statistical Normality Tests
# Shapiro-Wilk Test
from numpy.random import seed
from numpy.random import randn
from scipy.stats import shapiro
import statistics
from statistics import variance 
import matplotlib.pyplot as plt
import statsmodels.api as sm
from statsmodels.formula.api import ols
import seaborn as sns
import statsmodels.stats.multicomp
from scipy.stats import chisquare
import scipy.stats as stats
from statsmodels.stats.proportion import proportions_ztest

To determine whether there is any significant difference in the diameter of the cutlet between two units

Ho=There is no significance difference in the diameter of the cutlet between two units

H1= There is a significance difference in the diameter of the cutlet between two units

In [4]:
cut = pd.read_csv("cutlets.csv")

In [7]:
cut_a=cut.rename(columns={"Unit A":"sampleA","Unit B":"sampleB"})
cut_a.describe()

sampleA    sampleB
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

Check weather the Sample data is normally distributed or not
with the help of shapiro Test



In [8]:
#normality test
stat,p=shapiro(cut)

In [9]:
print('statistics=%.3f.p=%.3f'%(stat,p))

statistics=0.976.p=0.204


As we got P=0.204, p>0.05 hence we can accept null hypothesis. here null hypothesis is saying that the cutlet data is Normal

# To find the Variance for the given sample data with the help of varience Tests

In [10]:
c=pd.DataFrame(cut)
cut_a=cut.rename(columns={"Unit A":"sampleA","Unit B":"sampleB"})

In [11]:
print("Variance of sample setA = % s" %(statistics.variance(cut_a.sampleA)))

Variance of sample setA = 0.08317945374789917


In [12]:
print("Variance of sample setB = % s" %(statistics.variance(cut_a.sampleB)))

Variance of sample setB = 0.11792400440336137


# Compute Test Statistic¶
using Two Tailed t test we can compute that weather to accept the null hypothesis or not

In [13]:
# normalityt test
c= stats.ttest_ind(cut_a.sampleA,cut_a.sampleB)
print(c)

Ttest_indResult(statistic=0.7228688704678061, pvalue=0.4722394724599501)


# interpretation
as the obtained p value is more than alpha value(0.05),P>0.05 we can accept Null hypothesis. As we observed there is no significant difference in the diameter of the cutlet between 2 units.